auto_backup/models/db_backup.py

```diff
@@ -252,8 +252,13 @@ def schedule_backup(self):
                                     sftp.unlink(file)
                     # Close the SFTP session.
                     sftp.close()
+                    s.close()
```

Comment:
Wouldn't this fail if the following block failed?
```
try:
     s = paramiko.SSHClient()
     s.set_missing_host_key_policy(paramiko.AutoAddPolicy())
     s.connect(ipHost, portHost, usernameLogin, passwordLogin, timeout=20)
     sftp = s.open_sftp()
except Exception as error:
    _logger.critical('Error connecting to remote server! Error: ' + str(error))
```

Reply:
If the connection block fails, the sftp connection will fail too and the actual fail will be from sftp.chdir(pathToWriteTo):

```
2020-08-12 13:17:39,613 17171 INFO ? paramiko.transport: Connected (version 2.0, client OpenSSH_7.6p1) 
2020-08-12 13:17:41,760 17171 INFO ? paramiko.transport: Authentication (password) failed. 
2020-08-12 13:17:41,760 17171 CRITICAL xxx odoo.addons.auto_backup.models.db_backup: Error connecting to remote server! Error: Authentication failed. 
2020-08-12 13:17:41,760 17171 ERROR xxx odoo.addons.auto_backup.models.db_backup: Exception! We couldn't back up to the FTP server. Here is what we got back instead: local variable 'sftp' referenced before assignment 
```

So the change in this PR doesn't do any more harm, but of course it would actually make more sense to quit the backup process altogether if the connection fails, ie. to add something like return False to the connection exception handler. I can of course add that to this PR too if you want.

Reply:
Yeah we should already handle this one as it is much faster in the chain I guess:
`2020-08-12 13:17:41,760 17171 ERROR xxx odoo.addons.auto_backup.models.db_backup: Exception! We couldn't back up to the FTP server. Here is what we got back instead: local variable 'sftp' referenced before assignment `

Reply:
Yep. Though I noticed that if we return False from the connection exception, the last code block that removes any local files if necessary also doesn't get executed. If we want a better solution I think a more major rewrite might be warranted.

Reply:
Yep, you're right. I'm thinking of doing a major overhaul in V14 as I wrote this when I was less experienced & kind of kept forward porting. Let's go with your fix for this.